az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -376,9 +378,57 @@ public static void proxyUserKillAllSpawnedHadoopJobs(
         final UserGroupInformation proxyUser =
             HadoopSecureWrapperUtils.setupProxyUserWithHSM(hadoopSecurityManager, properties,
                 tokenFile.getAbsolutePath(), log);
+
+        // todo: print tokens in proxyUser
+        log.debug("proxyUserKillAllSpawnedHadoopJobs.proxyUser = " + proxyUser);
+        for (Token<?> token : proxyUser.getCredentials().getAllTokens()) {
+          log.debug(String.format("proxyUserKillAllSpawnedHadoopJobs.proxyUser.Token = %s, %s",
+              token.getKind(), token.getService()));
+        }
+        log.debug("proxyUserKillAllSpawnedHadoopJobs.proxyUser.Token --- end");
+
+        UserGroupInformation ugi0 = UserGroupInformation.getCurrentUser();
+        log.debug("proxyUserKillAllSpawnedHadoopJobs.getCurrentUser = " + ugi0);
+        for (Token<?> token : ugi0.getCredentials().getAllTokens()) {
+          log.debug(String.format("proxyUserKillAllSpawnedHadoopJobs.getCurrentUser.Token = %s, %s",
+              token.getKind(), token.getService()));
+        }
+        log.debug("proxyUserKillAllSpawnedHadoopJobs.getCurrentUser.Token --- end");
+
+        ugi0 = UserGroupInformation.getLoginUser();
+        log.debug("proxyUserKillAllSpawnedHadoopJobs.getLoginUser = " + ugi0);
+        for (Token<?> token : ugi0.getCredentials().getAllTokens()) {
+          log.debug(String.format("proxyUserKillAllSpawnedHadoopJobs.getLoginUser.Token = %s, %s",
+              token.getKind(), token.getService()));
+        }
+        log.debug("proxyUserKillAllSpawnedHadoopJobs.getLoginUser.Token --- end");
+
         proxyUser.doAs(new PrivilegedExceptionAction<Void>() {
           @Override
           public Void run() throws Exception {
+            log.debug("doAs.run.proxyUser = " + proxyUser);
+            for (Token<?> token : proxyUser.getCredentials().getAllTokens()) {
+              log.debug(String.format("doAs.run.proxyUser.Token = %s, %s", token.getKind(),
+                  token.getService()));
+            }
+            log.debug("doAs.run.proxyUser.Token --- end");
+
+            UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
+            log.debug("doAs.run.getCurrentUser = " + ugi);
+            for (Token<?> token : ugi.getCredentials().getAllTokens()) {
+              log.debug(String.format("doAs.run.getCurrentUser.Token = %s, %s", token.getKind(),
+                  token.getService()));
+            }
+            log.debug("doAs.run.getCurrentUser.Token --- end");
+
+            ugi = UserGroupInformation.getLoginUser();
```

Comment:
Is this the line fixing the issue ?
Could you please summarize what the root-cause was and how it's being addressed with this change?

Reply:
no, it's not, let me add comments and ping you there